src/objects.js

```diff
@@ -2837,55 +2710,162 @@ SpriteMorph.prototype.blockTemplates = function (category) {
         }
 
-    /////////////////////////////////
-
-        blocks.push('=');
-
         if (StageMorph.prototype.enableCodeMapping) {
+            blocks.push('=');
             blocks.push(block('doMapCodeOrHeader'));
             blocks.push(block('doMapValueCode'));
             blocks.push(block('doMapListCode'));
             blocks.push('-');
             blocks.push(block('reportMappedCode'));
-            blocks.push('=');
         }
+    }
+
+    return blocks;
+};
+
+// Utitlies displayed in the palette
+SpriteMorph.prototype.makeVariableButton = function () {
+    let button, myself = this;
+
+    function addVar(pair) {
+        var ide;
+        if (pair) {
+            if (myself.isVariableNameInUse(pair[0], pair[1])) {
+                myself.inform('that name is already in use');
+            } else {
+                ide = myself.parentThatIsA(IDE_Morph);
+                myself.addVariable(pair[0], pair[1]);
+                myself.toggleVariableWatcher(pair[0], pair[1]);
+                ide.flushBlocksCache('variables'); // b/c of inheritance
+                ide.refreshPalette();
+                ide.recordUnsavedChanges();
+            }
+        }
+    }
+
+    button = new PushButtonMorph(
+        null,
+        function () {
+            new VariableDialogMorph(
+                null,
+                addVar,
+                myself
+            ).prompt(
+                'Variable name',
+                null,
+                myself.world()
+            );
+        },
+        'Make a variable'
+    );
+    button.userMenu = this.helpMenu;
+    button.selector = 'addVariable';
+    button.showHelp = BlockMorph.prototype.showHelp;
+    return button;
+}
+
+SpriteMorph.prototype.deleteVariableButton = function () {
+    let button, myself = this;
+    button = new PushButtonMorph(
+        null,
+        function () {
+            var menu = new MenuMorph(
+                myself.deleteVariable,
+                null,
+                myself
+            );
+            myself.deletableVariableNames().forEach(name =>
+                menu.addItem(
+                    name,
+                    name,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    true // verbatim - don't translate
+                )
+            );
+            menu.popUpAtHand(myself.world());
+        },
+        'Delete a variable'
+    );
+    button.userMenu = this.helpMenu;
+    button.selector = 'deleteVariable';
+    button.showHelp = BlockMorph.prototype.showHelp;
+    return button;
+}
+
+SpriteMorph.prototype.devModeText = function () {
+    let txt = new TextMorph(localize('development mode \ndebugging primitives:'));
+    txt.fontSize = 9;
+    txt.setColor(this.paletteTextColor);
+    return txt;
+}
+
+SpriteMorph.prototype.helpMenu = function () {
```

Comment:
I don't know if this really belongs to a `SpriteMorph`, but this seems like a useful utility function throughout snap... there's a few other places where this is the entire context menu, I think.

Reply:
yeah, this is an excellent idea!